examples/vision/super_resolution_sub_pixel.py

```diff
@@ -0,0 +1,408 @@
+"""
+
+import PIL
+
+
+def plot_results(img, prefix, title):
+    """Plot the result with zoom-in area."""
+    img_array = img_to_array(img)
+    img_array = img_array.astype("float32") / 255.0
+
+    # Create a new figure with a default 111 subplot.
+    fig, ax = plt.subplots()
+    im = ax.imshow(img_array[::-1], origin="lower")
+
+    plt.title(title)
+    # zoom-factor: 2.0, location: upper-left
+    axins = zoomed_inset_axes(ax, 2, loc=2)
+    axins.imshow(img_array[::-1], origin="lower")
+
+    # Specify the limits.
+    x1, x2, y1, y2 = 200, 300, 100, 200
+    # Apply the x-limits.
+    axins.set_xlim(x1, x2)
+    # Apply the y-limits.
+    axins.set_ylim(y1, y2)
+
+    plt.yticks(visible=False)
+    plt.xticks(visible=False)
+
+    # Make the line.
+    mark_inset(ax, axins, loc1=1, loc2=3, fc="none", ec="blue")
+    plt.savefig(str(prefix) + "-" + title + ".png")
+    plt.show()
+
+
+"""
+Build `upscale_image`, `get_resized_image` and `get_lowres_image` methods to process image.
+"""
+
+
+def upscale_image(model, img):
+    """Predict the result based on input image and restore the image as RGB."""
+    ycbcr = img.convert("YCbCr")
+    y, cb, cr = ycbcr.split()
+    y = img_to_array(y)
+    y = y.astype("float32") / 255.0
+
+    input = np.expand_dims(y, axis=0)
+    out = model.predict(input)
+
+    out_img_y = out[0]
+    out_img_y *= 255.0
+
+    # Restore the image in RGB color space.
+    out_img_y = out_img_y.clip(0, 255)
+    out_img_y = out_img_y.reshape((np.shape(out_img_y)[0], np.shape(out_img_y)[1]))
+    out_img_y = PIL.Image.fromarray(np.uint8(out_img_y), mode="L")
+    out_img_cb = cb.resize(out_img_y.size, PIL.Image.BICUBIC)
+    out_img_cr = cr.resize(out_img_y.size, PIL.Image.BICUBIC)
+    out_img = PIL.Image.merge("YCbCr", (out_img_y, out_img_cb, out_img_cr)).convert(
+        "RGB"
+    )
+
+    return out_img
+
+
+def get_resized_image(img, w, h):
+    """"Return high-resolution original image"""
+    highres_img = img.resize((w, h))
+    return highres_img
+
+
+def get_lowres_image(img, upscale_factor):
+    """Return low-resolution image and will use it as input."""
+    lowres_img = img.resize(
+        (img.size[0] // upscale_factor, img.size[1] // upscale_factor),
+        PIL.Image.BICUBIC,
+    )
+    return lowres_img
+
+
+"""
+### Define callbacks to monitor training
+
+PSNR: We use PSNR to evaluate our model and please check it from
+[here](https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio) for more details.
+"""
+
+
+class ESPCNCallback(keras.callbacks.Callback):
+    def __init__(self):
+        super(ESPCNCallback, self).__init__()
+        self.psnr_tracker = keras.metrics.Mean("psnr")
+        self.test_img = get_lowres_image(load_img(test_img_paths[0]), upscale_factor)
+
+    # Store PSNR value in each epoch.
+    def on_epoch_begin(self, epoch, logs=None):
+        self.psnr = []
+
+    def on_epoch_end(self, epoch, logs=None):
+        self.psnr_tracker.update_state(self.psnr)
```

Comment:
You're using the Mean metric as if it were just a call to `np.mean()`. If you want to do that, then you don't need the mean metric at all, you can just use `np.mean()`.

Otherwise, you'd call update_state on the current scalar value of psnr in `on_batch_end`, and query results (and reset state) in on_epoch_end.

Reply:
Yeah, I think I can keep `np.mean`.